addons/mail/static/src/core/common/chat_window_service.js

```diff
@@ -32,7 +32,12 @@ export class ChatWindowService {
     }
 
     notifyState(chatWindow) {
-        if (this.ui.isSmall || !this.store.self || chatWindow.thread?.isTransient) {
+        if (
+            this.ui.isSmall ||
+            !this.store.self ||
+            chatWindow.thread?.isTransient ||
+            !chatWindow.thread?.hasSelfAsMember
```

Comment:
Hello, there are other issues in this flow, such as having no channel name in messaging menu or in chat window header.

messaging menu (missing title)
![image](https://github.com/odoo/odoo/assets/38213504/3358dd57-1c5e-41ea-be07-23e516c3ad8e)

inbox (missing from)
![image](https://github.com/odoo/odoo/assets/38213504/98834ebc-f6b2-4def-a966-016daa95c725)

header (bad title)
![image](https://github.com/odoo/odoo/assets/38213504/1fa400c8-f343-4e80-aed3-5927bf6bfdf5)

The condition here on `hasSelfAsMember` cannot possibly be true because `fetchChannelInfoState` is `not_fetched` at this step.

This would prevent from sending the proper update to the server if the current user is actually member of the channel in a same kind of flow (channel not yet fetched).

I'm thinking if we fetch or receive an inbox notification/message for a channel, we should ensure to call `getOrFetch` before anything else.
If binding to inbox is not easy or appropriate, worse case scenario we should call `getOrFetch` before opening the chat window, similarly as what we do when opening discuss.

But it seems already too late and it would not fix the missing name in messaging menu.

Reply:
I am not sure if we have any scenario where if you are a member of a channel than the channel won't be fetched. Both systray and discuss app would fetch all the required channels. So, I think it's ok to rely on this condition here.

Secondly, all the above issues you mentioned are already there before this PR. So, I think above issues should be handled separately in this [task](https://www.odoo.com/web?debug=1#id=3890896&cids=2&menu_id=4722&action=333&active_id=1519&model=project.task&view_type=form)